### DISCUSSION TITLE
Is there any way to ignore folders in pages and api folder?

### DISCUSSION THREAD

**Discussion by darwin403** [vercel/next.js#18743] [Help] [answered]
As our application is scaling, we are using pages merely as "**routers**" with all components, utilities and tests moved into another containers folder. 

I think this defeats the major feature of Next.js wherein you can create a completely self-contained page in a folder rather than abstract it out to other folders.

Does not know how ignoring special directories like `pages/myroute/_components` or `pages/api/myroute/_utils` can be achieved?

Look forward to your feedback!

**Comment by jamesmosier** [comment, marked as answer]
Hi @darwin403. There is no way to ignore any folders or files within the `pages` directory. 

The only option would be to disable filesystem routing altogether via [`useFileSystemPublicRoutes`](https://nextjs.org/docs/advanced-features/custom-server#disabling-file-system-routing) but that would require a custom server and setting up your routes in there.
